modules/rgbd/src/tsdf.cpp

```diff
@@ -627,7 +629,10 @@ inline Point3f TSDFVolumeCPU::getNormalVoxel(Point3f p) const
         nv = v0 + tx*(v1 - v0);
     }
 
-    return normalize(an);
+    float nv = sqrt(an[0] * an[0] +
+        an[1] * an[1] +
+        an[2] * an[2]);
+    return nv < 0.0001 ? nan3 : an / nv;
```

Comment:
Please fix here and everywhere it's used: `0.0001` constant is of double type, comparison between `float nv` and double constant is a source of warnings; use `0.0001f` instead.

Reply:
done